Readd "Riese/hide columns""

MartinRiese opened the pull request:
Reverts dimagi/commcare-hq#36141 to get https://github.com/dimagi/commcare-hq/pull/35849 a back plus a couple of commits

## Product Description

https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/edit-v2/2994667666?draftShareId=a9c7feeb-f53e-44d8-9af6-f3ad17726758

![2025-03-21_13-30-31 (1)](https://github.com/user-attachments/assets/00f6d2d7-8790-4e76-b1a4-d3f1c2077372)

## Technical Summary

https://dimagi.atlassian.net/browse/USH-5319

## Feature Flag

No

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

MartinRiese committed:
Revert "Revert "Riese/hide columns""

MartinRiese committed:
Account for search first

Jtang-1 commented on corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js:
```diff
@@ -683,7 +683,11 @@ hqDefine("cloudcare/js/formplayer/menus/views", [
+                if (attributes.styles) {
+                    this.set('columnCanBeVisible', attributes.styles.map(s => s.widthHint !== 0));
+                } else {
+                    this.set('columnCanBeVisible', Array(attributes.columnNames.length).fill(true));
```
i'm not understanding what the cause of the bug was and how this fixes it?

MartinRiese replied:
The navigate_menu response for search first does not include any styles. So `attributes.styles` was `undefined`. After the search the view is rebuild and `columnCanBeVisible` get populated correctly. I could probably have just assigned and empty array but I thought creating the right size array might prevent some other errors that I have not come across yet.

MartinRiese committed:
Get hex code from sha512 hash

MartinRiese committed:
Do not store column name

Column names should be reloaded in case users change languages.
Use can be visible instead

MartinRiese committed:
Add quotes around empty column names

MartinRiese committed:
Woody says no quotes

MartinRiese committed:
Fix tests

MartinRiese committed:
Fix test the proper way

MartinRiese committed:
Fix lint errors
